pkg/operator/resources/batchapi/api.go

```diff
@@ -77,11 +77,19 @@ func UpdateAPI(apiConfig *userconfig.API, projectID string) (*spec.API, string,
 
 		err = applyK8sResources(api, prevVirtualService)
 		if err != nil {
-			return nil, "", err
+			rollBackErr := rollBack(prevVirtualService)
+			if rollBackErr != nil {
+				return nil, "", err
+			}
+			return nil, "update failed; rollbacked to previous deployment", nil
```

Comment:
I think it might still be good to return the error, in case the user can do anything about it